src/main/java/uk/gov/cslearning/catalogue/service/upload/FileFactory.java

```diff
@@ -2,12 +2,11 @@
 
 import org.springframework.stereotype.Component;
 
-import java.io.File;
-import java.net.URISyntaxException;
+import java.io.InputStream;
 
 @Component
 public class FileFactory {
-    public File get(String path) throws URISyntaxException {
-        return new File(getClass().getResource(path).toURI());
+    public InputStream getInputStreamFromPath(String path) {
```

Comment:
Given that this no longer returns a File it probably shouldn't be in FileFactory. There's an InputStreamFactory class it would make more sense in there.